azerothcore/azerothcore-wotlk
# Core/Social: cleaunup, fix crash when friend or ignore get 0 guid and also double friend list

## CHANGES PROPOSED:
-  cleanup
-  fix crash when friend or ignore get 0 guid( https://paste2.org/3dUy2Ud8 )
-  fix floating problem with double friend list(thx to TC team)

## ISSUES ADDRESSED:
don't find

## TESTS PERFORMED:
- build and game

## How to test the changes

-  Play with the Friend system (add/remove friends to your friend list), try login and logout between these actions too, make sure everything is fine

## Target branch(es):

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
